sdk/formrecognizer/ai-form-recognizer/src/lro/util/delayMs.ts

```diff
@@ -37,30 +38,38 @@ export interface CancelablePromiseLike<T> extends PromiseLike<T> {
  * @param ms - the number of milliseconds to wait before resolving
  * @param cb - a callback that can provide the caller with a cancellation function
  */
-export function delayMs(ms: number): CancelablePromiseLike<void> {
+export function delayMs(
+  ms: number,
+  abortSignal: AbortSignalLike | undefined
+): CancelablePromiseLike<void> {
   let aborted = false;
-  let toReject: (() => void) | undefined;
+  let toReject: ((e: Error) => void) | undefined;
+
+  abortSignal?.addEventListener("abort", () => {
+    aborted = true;
+    toReject?.(new Error("The operation was aborted."));
+  });
 
   return Object.assign(
     new Promise<void>((resolve, reject) => {
       let token: CancellationToken | undefined;
       toReject = () => {
         maybemap(token, clearTimeout);
-        reject(INTERRUPTED);
+        reject(new Error(INTERRUPTED));
       };
 
       // In the rare case that the operation is _already_ aborted, we will reject instantly. This could happen, for
       // example, if the user calls the cancellation function immediately without yielding execution.
       if (aborted) {
-        toReject();
+        toReject(new Error("The operation was canceled prematurely."));
```

Comment:
Should we throw `AbortError` instead?

Reply:
I considered it and I do throw `AbortError` if the `abortSignal` was aborted, but this one happens when the operation is canceled prematurely, i.e. a user instantiates the poller, calls `pollUntilDone` on it, and then `cancel`'s the poller before it can even start waiting. I'm pretty sure that is _somehow_ an error codepath. Since the abortSignal might not be involved here and I don't really know how this is reachable I decided to throw a normal error. Open to the idea of doing an AbortError, though, if you feel it's compelling.

Reply:
I think there is something wrong with the logic here. I'm going to rework it a little bit.

Reply:
Shameless plug: May I suggest using core-util's `createAbortablePromise`? 😊

Reply:
Ooooh. Is that released already?

Reply:
It indeed is!